React/Base/macOS/RCTUIKit.m

```diff
@@ -251,6 +251,23 @@ - (instancetype)initWithCoder:(NSCoder *)coder
   return RCTUIViewCommonInit([super initWithCoder:coder]);
 }
 
+- (BOOL)acceptsFirstMouse:(NSEvent *)event
+{
+  if (self.acceptsFirstMouse || [super acceptsFirstMouse:event]) {
+    return YES;
+  }
+
+  // If any RCTUIView view above has acceptsFirstMouse set, then return YES here.
+  NSView *view = self;
+  while ((view = view.superview)) {
```

Comment:
This logic kinda surprises me

If you have a wrapper view around the screen, and then a child button that you explicitly set `acceptsFirstMouse={false}`, it sounds like this will find the parent view which is set to true and then override the value on the button? Am I reading the logic wrong?

Reply:
I guess the thinking here is it will be more intuitive for users to write something like the following? Which I would agree with.

```js
<Touchable acceptsFirstMouse>
  <Image />
<Touchable>
```

Whereas it seems less likely that a component will explicitly opt-out of the behaviour? If we feel strongly about being able to do that then there probably needs to be another property that can represent explicit vs default? (Or back the property with a different type of variable and convert to `BOOL` in the setter/getter such that we can add a predicate method to check for explicit vs default.)

Reply:
I was assuming that the default would be to accept first mouse, but let people disable that for buttons that take destructive actions. I think that's how Scott described the Apple HIG to me a while back

Reply:
Exactly as @alloy says. I can't think of conceivable circumstance where a child view of a pressable view would want to opt-out. I can only imagine that result in bugs and bad UX. 😀

Reply:
The default behaviour is that no view accepts first mouse, but I agree that it seems an edge-case to want a subview to not accept it. If/when that case surfaces we can still easily introduce the explicit vs default disablement detection.

Reply:
Ok, so I take it this is resolved then?